### PR TITLE
refactor chm_htmlescape()

### PR DESCRIPTION
**Pull request by tk0miya**
### Feature or Bugfix
- Refactoring

### Purpose
- refs: https://github.com/sphinx-doc/sphinx/pull/5853#discussion_r243767436

### CODE REVIEW

File: tests/test_build_htmlhelp.py [hunk 1]
```diff
@@ -22,5 +26,28 @@ def test_chm(app):
...
+        assert chm_htmlescape('<Hello> "world"', True) == '&lt;Hello&gt; &quot;world&quot;'
+        assert chm_htmlescape('<Hello> "world"', False) == '&lt;Hello&gt; "world"'
+    else:
+        assert chm_htmlescape('<Hello> "world"') == '&lt;Hello&gt; &quot;world&quot;'
+        assert chm_htmlescape('<Hello> "world"', True) == '&lt;Hello&gt; &quot;world&quot;'
+        assert chm_htmlescape('<Hello> "world"', False) == '&lt;Hello&gt; "world"'
 
+    if PY2:
+        assert chm_htmlescape("Hello 'world'") == "Hello 'world'"
+        assert chm_htmlescape("Hello 'world'", True) == "Hello &#39;world&#39;"
```
Comment (tk0miya) [inline tests/test_build_htmlhelp.py]: This is new behavior on py2. At present, it works only for .hhc file.

Comment (tk0miya) [reply tests/test_build_htmlhelp.py]: I just reverted this. It's too much.

**Comment by ghost** [comment]
Maybe we should also change `title = chm_htmlescape(title)` to `title = chm_htmlescape(title, True)` at line 332:
https://github.com/sphinx-doc/sphinx/pull/5862/files#diff-934a4ea68760fc7303dd44a301179b73R332

Imagine someone building a .chm using Python2, and a quote `"` appears in title, this seems a bug in this situation.

**Comment by tk0miya** [comment]
@animalize Absolutely!

File: sphinx/builders/htmlhelp.py [hunk 2]
```diff
@@ -327,10 +326,9 @@ def write_index(title, refs, subitems):
                 # type: (unicode, List[Tuple[unicode, unicode]], List[Tuple[unicode, List[Tuple[unicode, unicode]]]]) -> None  # NOQA
                 def write_param(name, value):
                     # type: (unicode, unicode) -> None
-                    item = '    <param name="%s" value="%s">\n' % \
-                        (name, value)
+                    item = '    <param name="%s" value="%s">\n' % (name, value)
                     f.write(item)
-                title = chm_htmlescape(title)
+                title = chm_htmlescape(title, True)
```
Comment (tk0miya) [inline sphinx/builders/htmlhelp.py:331]: I add 2nd argument to fix the bug. see CHANGES.

**Comment by ghost** [comment]
I put this `htmlhelp.py` into `X:\Python37\Lib\site-packages\sphinx\builders`, then I can build a correct `python372rc1.chm`.

**Comment by tk0miya** [comment]
Thank you for reviewing! Merging now.

**Pull request merged**